sync/mock.go

```diff
@@ -0,0 +1,56 @@
+package sync
+
+import (
+	"github.com/weaveworks/flux"
+	"github.com/weaveworks/flux/policy"
+)
+
+type rsc struct {
+	bytes []byte
+	Kind  string
+	Meta  struct {
+		Namespace string
+		Name      string
+	}
+}
+
+type rscIgnorePolicy struct {
+	rsc
+}
+
+func (rs rsc) Source() string {
+	return ""
+}
+
+func (rs rsc) Bytes() []byte {
+	return []byte{}
+}
+
+func (rs rsc) ResourceID() flux.ResourceID {
+	return flux.MakeResourceID(rs.Meta.Namespace, rs.Kind, rs.Meta.Name)
+}
+
+func (rs rsc) Policy() policy.Set {
+	p := policy.Set{}
+	return p
+}
+
+func (ri rscIgnorePolicy) Policy() policy.Set {
+	p := policy.Set{}
+	p[policy.Ignore] = "true"
+	return p
+}
+
+func mockResourceWithoutIgnorePolicy(kind, namespace, name string) rsc {
+	r := rsc{Kind: kind}
+	r.Meta.Namespace = namespace
+	r.Meta.Name = name
+	return r
+}
+
+func mockResourceWithIgnorePolicy(kind, namespace, name string) rscIgnorePolicy {
```

Comment:
You can de-duplicate some code here with:
```
return rscIgnorePolicy{mockResourceWithoutIgnorePolicy(kind, namespace, name)}
```

Reply:
Thanks for the tip. Will remember it :+1:  I have spent quite a lot of time on this issue and if I change, I shall need to test, squash the commits. I have just managed to tame all the commits into one, at last (problems with rebasing), so eager to get rid of this. Next time is just wound the corner.